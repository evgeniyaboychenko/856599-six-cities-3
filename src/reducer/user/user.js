import {extend} from '../../utils/utils.js';
import {adaptUser} from '../../utils/adapter.js';
import {apiDefault} from '../../api.js';
const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  error: ``,
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  user: {
    id: -1,
    name: ``,
    avatar: ``,
    email: ``,
    isPro: false,
  }
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_LOADING_ERROR: `SET_LOADING_ERROR`
};

const ActionCreator = {
  requireAuthorization: (status, user) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: {status, user}
    };
  },
  setLoadingError: (error) => {
    return {
      type: ActionType.SET_LOADING_ERROR,
      payload: error
    };
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload.status,
        user: action.payload.user
      });
    case ActionType.SET_LOADING_ERROR:
      return extend(state, {
        error: action.payload
      });
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch) => {
    return apiDefault.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, adaptUser(response.data)));
      }
      // .catch((err) => {
      //   throw err;
      // });
      , () => dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH, {id: -1,
        name: ``,
        avatar: ``,
        email: ``,
        isPro: false})));
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, adaptUser(response.data)));
      })
    .catch((err) => {
      const {response} = err;
      dispatch(ActionCreator.setLoadingError(response.data.error));
      // throw err;
    });
  },
};

export {reducer, ActionCreator, ActionType, AuthorizationStatus, Operation};
