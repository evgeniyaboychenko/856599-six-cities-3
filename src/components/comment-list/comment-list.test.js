import React from 'react';
import renderer from 'react-test-renderer';
import CommentList from './comment-list.jsx';
import {AuthorizationStatus} from '../../reducer/user/user.js';

jest.mock(`../comment-form/comment-form.jsx`);

const DESCRIPTIONS = [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`];
const AVATARS = [`avatar-angelina.jpg`, `avatar-max.jpg`];
const NAMES = [`Angelina`, `Max`, `Kate`, `Pol`, `Mike`];
const comments =
[{
  id: 1,
  avatar: AVATARS[0],
  name: NAMES[0],
  rating: 2,
  text: DESCRIPTIONS[0],
  date: `2020-03-29T18:07:19.382Z`,
  idUser: 1,
  isPro: false,
},
{
  id: 2,
  avatar: AVATARS[1],
  name: NAMES[1],
  rating: 2,
  text: DESCRIPTIONS[1],
  date: `2020-03-29T18:07:19.382Z`,
  idUser: 1,
  isPro: false,
}];

const authorizationStatus = AuthorizationStatus.AUTH;

it(`should CommentList render correctly`, () => {
  const tree = renderer.create(
      <CommentList
        comments = {comments}
        idCard = {1}
        authorizationStatus = {authorizationStatus}
        isSubmitForm = {false}
        errorForm = {``}
      />).toJSON();
  expect(tree).toMatchSnapshot();
});
