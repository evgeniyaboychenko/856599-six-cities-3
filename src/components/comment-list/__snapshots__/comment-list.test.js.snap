// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should CommentList render correctly 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Angelina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": 40,
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
        </p>
        <time
          className="reviews__time"
          dateTime="March 2020"
        >
          March 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": 40,
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
        </p>
        <time
          className="reviews__time"
          dateTime="March 2020"
        >
          March 2020
        </time>
      </div>
    </li>
  </ul>
</section>
`;
